client/src/Components/FormComboBox.js

```diff
@@ -6,36 +6,70 @@ import { comboBox } from '../../node_modules/uswds/src/js/components';
 import { ValidationState } from '../Models/PatientFieldData';
 import ValidationMessage from './ValidationMessage';
 
+/**
+ * Normally, the USWDS ComboBox ONLY allows selection of a value from the options list. The
+ * text input is used only as a typeahead search filter.
+ *
+ * This component attempts to extend the behavior of the ComboBox such that it also allows
+ * a free-form value typed in the text input box to be an acceptable value. For example,
+ * if our CAD integration is behind and an incident number is not yet available in the dropdown,
+ * it can still be typed in manually into the box.
+ *
+ */
 function FormComboBox({ label, property, required, onChange, options, showRequiredHint, size, validationState, value }) {
   const ref = useRef();
   const [focused, setFocused] = useState(false);
-  const [customOption, setCustomOption] = useState(null);
+  // the inputValue state mirrors the contents of the input box created by USWDS ComboBox
+  const [inputValue, setInputValue] = useState(value);
 
   useEffect(() => {
     const { current } = ref;
     comboBox.on(current);
-    // manually add event handlers to the custom input added by USWDS
+    // get a reference to the select component so we can dispatch events through it
     const select = current.querySelector('select');
-    const input = current.querySelector('input[type="text"]');
-    input.addEventListener('input', (e) => setCustomOption(e.target.value));
-    input.addEventListener('focus', () => setFocused(true));
-    input.addEventListener('blur', (e) => {
-      setFocused(false);
-      select.value = e.target.value;
+    function dispatchEvent(value) {
+      select.value = value;
       const event = new CustomEvent('change', {
         bubbles: true,
         cancelable: true,
         detail: { value },
       });
       select.dispatchEvent(event);
+    }
+    // manually add event handlers to the custom input added by USWDS
+    const input = current.querySelector('input[type="text"]');
```

Comment:
I guess you can't just store a reference to the input because it doesn't exist until after the USWDS scripts run?

Reply:
That's right, the input is created by the USWDS javavascript when it is turned "on"...